frontend/src/pages/Main.tsx

```diff
@@ -185,6 +186,97 @@ const Main: FC = () => {
     });
   }, []);
 
+  /**
+    /* Function to calculate the maximum distance
+    /*
+    /* @author Chen
+    */
+  const calculateMaxDistance = (selectedRoad: IRoad): number => {
+    const wayIds = Object.keys(selectedRoad.geometries);
+    let maxDistance = 0;
+    let farthestCoordinate: L.LatLng | null = null;
+
+    // Convert all coordinates to Leaflet LatLng objects and store in an array
+    const allLeafletCoordinates: L.LatLng[] = wayIds.flatMap((wayId) =>
+      selectedRoad.geometries[wayId].map(
+        (coord) => L.latLng(coord.lat, coord.lng), // Use the L.latLng function
+      ),
+    );
+    // If there are no coordinates, exit early
+    if (allLeafletCoordinates.length === 0) {
+      return 0;
+    }
+    // Initialize farthestCoordinate with the first coordinate, if available
+    if (allLeafletCoordinates.length > 0) {
+      farthestCoordinate = allLeafletCoordinates[0];
+    }
+    // Find the point farthest from the first coordinate
+    const firstLeafletCoordinate = allLeafletCoordinates[0];
+    allLeafletCoordinates.forEach((leafletCoordinate) => {
+      const distance = firstLeafletCoordinate.distanceTo(leafletCoordinate);
+      if (distance > maxDistance) {
+        maxDistance = distance;
+        farthestCoordinate = leafletCoordinate;
+      }
+    });
+
+    // Reset max distance for the second iteration
+    maxDistance = 0;
+
+    // Find the point farthest from the farthest point found in step 1
+    if (farthestCoordinate) {
+      allLeafletCoordinates.forEach((leafletCoordinate) => {
+        const distance = farthestCoordinate!.distanceTo(leafletCoordinate);
+        if (distance > maxDistance) {
+          maxDistance = distance;
+        }
+      });
+    }
+
+    return maxDistance;
+  };
```

Comment:
Instead of finding the two farthest points, can't you just sum the size of each way? And have the length of the road?

Reply:
Sounds a good idea